app/src/main/java/com/minimalisticapps/priceconverter/presentation/ui/widget/TextInputShitCoin.kt

```diff
@@ -5,141 +5,32 @@ import androidx.compose.foundation.layout.padding
 import androidx.compose.foundation.text.KeyboardOptions
 import androidx.compose.material.TextField
 import androidx.compose.runtime.Composable
-import androidx.compose.runtime.mutableStateOf
-import androidx.compose.runtime.remember
+import androidx.compose.runtime.MutableState
 import androidx.compose.ui.Modifier
-import androidx.compose.ui.focus.onFocusChanged
-import androidx.compose.ui.text.TextRange
+import androidx.compose.ui.focus.onFocusEvent
 import androidx.compose.ui.text.input.KeyboardType
 import androidx.compose.ui.text.input.TextFieldValue
 import androidx.compose.ui.unit.dp
-import androidx.hilt.navigation.compose.hiltViewModel
-import com.minimalisticapps.priceconverter.common.utils.parseBigDecimalFromString
-import com.minimalisticapps.priceconverter.common.utils.toSatsFormat
-import com.minimalisticapps.priceconverter.presentation.home.viewmodels.HomeViewModel
-import com.minimalisticapps.priceconverter.room.entities.FiatCoinExchange
 
 @Composable
 fun TextInputShitCoin(
-    rate: Double?,
-    onValueChange: (String) -> Unit,
-    fiatCoinExchange: FiatCoinExchange,
-    viewModel: HomeViewModel = hiltViewModel(),
+    state: MutableState<TextFieldValue>,
+    onValueChange: (TextFieldValue) -> Unit,
+    onSelected: () -> Unit
 ) {
-    val searchText = remember {
-        mutableStateOf(
-            TextFieldValue(
-                fiatCoinExchange.shitCoinValue,
-                TextRange(fiatCoinExchange.shitCoinValue.length)
-            )
-        )
-    }
-    val isFocused = remember {
-        mutableStateOf(false)
-    }
-
-    val count = remember {
-        mutableStateOf(0)
-    }
-
-    if (viewModel.textFieldValueBtc.value.text.isNotEmpty() && !isFocused.value) {
```

Comment:
the model manipulation was moved to model, component is responsible only for UI now

This is significant reduction of complexity and code. Also follows SRP and allows easy testing.